### PR TITLE
Add "/" to valid characters for instrument names

### LINKED ISSUE #3422: Allow metric instrument names to contain `/` characters
**What are you trying to achieve?**

Create instruments/metrics with names like `composer.googleapis.com/environment/database/auto_failover_request_count`:

- containing `/` characters
- exceeded 63 characters.

These types of metrics are very common in Google Cloud Monitoring ([examples](https://cloud.google.com/monitoring/api/metrics_gcp), notice the domain prefix is part of the metric) and we would like it to be possible to create such metrics with OTel instrumentation.

**What did you expect to see?**

It should be possible to create such an instrument e.g. in Python
```py
from opentelemetry.sdk.metrics import MeterProvider
MeterProvider().get_meter("foo").create_counter("composer.googleapis.com/environment/database/auto_failover_request_count")
```
However it fails with
```py
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/google/home/aaronabbott/repo/opentelemetry-js/venvtmp/lib/python3.10/site-packages/opentelemetry/sdk/metrics/_internal/__init__.py", line 97, in create_counter
    instrument = _Counter(
  File "/usr/local/google/home/aaronabbott/repo/opentelemetry-js/venvtmp/lib/python3.10/site-packages/opentelemetry/sdk/metrics/_internal/instrument.py", line 56, in __init__
    raise Exception(_ERROR_MESSAGE.format(name))
Exception: Expected ASCII string of maximum length 63 characters but got composer.googleapis.com/environment/database/auto_failover_request_count
```

[As required by the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/metrics/api.md#instrument-name-syntax)

> - Subsequent characters must belong to the alphanumeric characters, '_', '.', and '-'.
> - They can have a maximum length of 63 characters.

**Additional context.**

Looks like this was introduced in https://github.com/open-telemetry/opentelemetry-specification/pull/1557 by @reyang and the oldest reference I could find to these requirements was in https://github.com/open-telemetry/opentelemetry-specification/pull/601 by @jmacd.

### PR DESCRIPTION
**Pull request by aabmass**
Fixes #3422

## Changes

Updated the allowed characters for metric instrument names in the OTel API to permit the `/` character. See #3422 for use cases and discussion.

* [x] Related issues #3422
* [x] [`CHANGELOG.md`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/CHANGELOG.md) file updated for non-trivial changes

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: specification/metrics/api.md [hunk 1]
```diff
@@ -206,7 +206,7 @@ The instrument name syntax is defined below using the [Augmented Backus-Naur
 Form](https://tools.ietf.org/html/rfc5234):
 
 ```abnf
-instrument-name = ALPHA 0*254 ("_" / "." / "-" / ALPHA / DIGIT)
+instrument-name = ALPHA 0*254 ("_" / "." / "-" / "/" / ALPHA / DIGIT)
```
Comment (cijothomas) [inline specification/metrics/api.md:209]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1 I suppose there is no need of changes in the openmetrics compat part, as it already talks about what to do for unsupported characters, and hence it'll just replace "/" with "_", just like others unsupported chars like "." ?

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
"Instrument names conform to the specified syntax." This section from https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md should be reset or modified with a new entry, to make it easy to tell which languages have implemented this.

File: specification/metrics/api.md (hunk 1 above)
Comment (reyang) [reply specification/metrics/api.md:209]: +1 correct

**Review by utpilla — APPROVED** [review]

File: specification/metrics/api.md (hunk 1 above)
Comment (aabmass) [reply specification/metrics/api.md:209]: +1 as well, but would be good to get @gouthamve's thoughts on this.

**Comment by aabmass** [comment]
> "Instrument names conform to the specified syntax." This section from https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md should be reset or modified with a new entry, to make it easy to tell which languages have implemented this.

Thanks, I cleared out the row

File: specification/metrics/api.md (hunk 1 above)
Comment (reyang) [reply specification/metrics/api.md:209]: > Interesting, so at that point OTel would actually be less permissive? We should probably consider doing the same

+1

**Pull request merged**